src/OrleansRuntime/Counters/SiloStatisticsManager.cs

```diff
@@ -12,18 +12,23 @@ internal class SiloStatisticsManager
         internal SiloPerformanceMetrics MetricsTable;
         private readonly Logger logger = LogManager.GetLogger("SiloStatisticsManager");
 
-        internal SiloStatisticsManager(GlobalConfiguration globalConfig, NodeConfiguration nodeConfig)
+        public SiloStatisticsManager(SiloInitializationParameters initializationParams)
         {
             MessagingStatisticsGroup.Init(true);
             MessagingProcessingStatisticsGroup.Init();
             NetworkingStatisticsGroup.Init(true);
-            ApplicationRequestsStatisticsGroup.Init(globalConfig.ResponseTimeout);
+            ApplicationRequestsStatisticsGroup.Init(initializationParams.GlobalConfig.ResponseTimeout);
             SchedulerStatisticsGroup.Init();
             StorageStatisticsGroup.Init();
             runtimeStats = new RuntimeStatisticsGroup();
-            logStatistics = new LogStatistics(nodeConfig.StatisticsLogWriteInterval, true);
-            MetricsTable = new SiloPerformanceMetrics(runtimeStats, nodeConfig);
-            countersPublisher = new CountersStatistics(nodeConfig.StatisticsPerfCountersWriteInterval);
+            this.logStatistics = new LogStatistics(initializationParams.NodeConfig.StatisticsLogWriteInterval, true);
```

Comment:
It feels like these could be give as parameters in the future. As in not creating instances in constructor using `new` and configuration options, but giving the instances via constructor and moving the configurations closer to the source. Hene "unmasking" the process.